src/vm/siginfo.cpp

```diff
@@ -4953,11 +4953,54 @@ void PromoteCarefully(promote_func   fn,
     (*fn) (ppObj, sc, flags);
 }
 
+void ReportByRefPointersFromByRefLikeObject(promote_func *fn, ScanContext *sc, PTR_MethodTable pMT, PTR_VOID pSrc)
+{
+    WRAPPER_NO_CONTRACT;
+
+    _ASSERTE(pMT->IsByRefLike());
+
+    if (pMT == g_TypedReferenceMT)
```

Comment:
The TypedReference has not been allowed as a field and so there is no problem today. They have been only allowed as regular arguments. We should not come for regular TypedReference arguments at all because of they are special cased above ReportPointersFromValueType.

Since you have added the special case for TypedReference here, I thought you want to prepare this code for case where TypedReference are allowed as fields as other byref-like types. There is a good chance that TypedReferences are going to be allowed as fields as natural fallout from the relaxations of rules for byref-like types in C#.

Either of these two options would sound reasonable to me:
- Do not prepare this for TypedReference fields - do not handle TypedReference at all here
- Prepare this for TypedReference fields - handle TypedReference same way as `ByReference<T>` here